index.html

```diff
@@ -957,7 +957,13 @@ <h4>Variation: concatenating files of different resolutions</h4>
       <p><code>ffmpeg -i input1.avi -i input2.mp4 -filter_complex "[0:v:0] scale=1920:1080:flags=lanczos [rescaled_video], [rescaled_video] [0:a:0] [1:v:0] [1:a:0] concat=n=2:v=1:a=1 [video_out] [audio_out]" -map "[video_out]" -map "[audio_out]" <i>output_file</i></code></p>
       <p>However, this will only have the desired visual output if the inputs have the same aspect ratio. If you wish to concatenate an SD and an HD file, you will also wish to pillarbox the SD file while upscaling. (See the <a href="https://amiaopensource.github.io/ffmprovisr/#SD_HD_2">Convert 4:3 to pillarboxed HD</a> command). The full command would look like this:</p>
       <p><code>ffmpeg -i input1.avi -i input2.mp4 -filter_complex "[0:v:0] scale=1440:1080:flags=lanczos, pad=1920:1080:(ow-iw)/2:(oh-ih)/2 [to_hd_video], [to_hd_video] [0:a:0] [1:v:0] [1:a:0] concat=n=2:v=1:a=1 [video_out] [audio_out]" -map "[video_out]" -map "[audio_out]" <i>output_file</i></code></p>
-      <p>Here, the first input an SD file which needs to be upscaled to match the second input, which is 1920x1080. The scale filter enlarges the SD input to the height of the HD frame, keeping the 4:3 aspect ratio; then, the video is pillarboxed within a 1920x1080 frame.</p>
+      <p>Here, the first input is an SD file which needs to be upscaled to match the second input, which is 1920x1080. The scale filter enlarges the SD input to the height of the HD frame, keeping the 4:3 aspect ratio; then, the video is pillarboxed within a 1920x1080 frame.</p>
+      <h4>Variation: concatenating files of different framerates</h4>
+      <p>If the input files have different framerates, then the output file may be of variable framerate. To explicitly attain an output file of constant framerate, you may wish convert an input (or multiple inputs) to a different framerate prior to concatenation.</p>
+      <p>You can speed up or slow down a file using the <code>fps</code> and <code>atempo</code> filters (see also the <a href="https://amiaopensource.github.io/ffmprovisr/#modify_speed">Modify speed</a> command).</p>
+      <p>Here's an example of the full command, in which input1 is 30fps, input2 is 25fps, and 25fps is the desired output speed.</p>
+      <p><code>ffmpeg -i input1.avi -i input2.mp4 -filter_complex "[0:v:0] fps=fps=25 [video_to_25fps]; [0:a:0] atempo=(25/30) [audio_to_25fps]; [video_to_25fps] [audio_to_25fps] [1:v:0] [1:a:0] concat=n=2:v=1:a=1 [video_out] [audio_out]" -map "[video_out]" -map "[audio_out]" <i>output_file</i></code></p>
```

Comment:
I’m sorry, I would suggest `input_1`, etc.

Reply:
Question: if you concatenate `.avi` and `.mp4`, do you not need a parameter which enforces the higher quality. I guess, otherwise the lower is chosen, but I may be wrong.

Reply:
Will snake-case the names.   
I'm not totally clear on what you are asking @retokromer! Are you able to rephrase?

Reply:
@kfrn If one mixes different containers/codecs, without forcing the quality (e.g. with `-pix_fmt` for the image), then FFmpeg try to chose the best possible, but I have the **_feeling_** that sometimes it fails. I had a discussion about that some years ago with Nicolas George in Paris. I **_guess_** it’s more secure to impose. You can impose the higher quality to the lower and it might make the lower worst. You can impose the lower quality to the higher and you always will loose quality.

(I don’t have the time to delve into this in a scientific way now. Possibly you have already? I suggest to commit this PR. We can always change later, when needed.)

Reply:
I think I'm with you now ... so, unless you explicitly specify otherwise, it's the output file type which will determine the file quality. e.g., if you have `output.mp4` the video will be encoded as H.264 with the default settings ([i.e.](https://trac.ffmpeg.org/wiki/Encode/H.264), crf 23, preset medium).  

If you mean `-pix_fmt` specifically, I actually tested what happens if the two inputs have different chroma subsampling schemes - but I can't remember the result. :-/  I can probably test it tonight.

Reply:
*Update*: I didn't test things extensively, but when concat-ing a 4:2:2 and a 4:2:0 file, I'm observing 4:2:2 outputs.

Reply:
Yes, I guess we can merge it. I’ll test a little with WebM/VP8 later this week, as we have a client that wishes that combination and the amateur part of the collection contains different frame rates.

Reply:
Cool, please report back! :-)